src/Shared/PackageManagers/BaseNuGetProjectManager.cs

```diff
@@ -0,0 +1,98 @@
+        {
+            if (packageUrl is null || packageUrl.Qualifiers?.TryGetValue("repository_url", out string? repositoryUrlQualifier) != true ||
+                repositoryUrlQualifier?.TrimEnd('/') == NuGetProjectManager.NUGET_DEFAULT_INDEX)
+            {
+                return new NuGetProjectManager(destinationDirectory, NuGetPackageActions.CreateV3(), httpClientFactory, timeout);
+            }
+            else if (repositoryUrlQualifier?.TrimEnd('/') == NuGetV2ProjectManager.POWER_SHELL_GALLERY_DEFAULT_INDEX)
+            {
+                return new NuGetV2ProjectManager(destinationDirectory, NuGetPackageActions.CreateV2(), httpClientFactory, timeout);
+            }
+            else
+            {
+                throw new NotImplementedException($"NuGet package URLs having a repository URL other than '{NuGetV2ProjectManager.POWER_SHELL_GALLERY_DEFAULT_INDEX}' or '{NuGetProjectManager.NUGET_DEFAULT_INDEX}' are not currently supported.");
+            }
+        }
+
+        /// <inheritdoc />
+        public override async IAsyncEnumerable<PackageURL> GetPackagesFromOwnerAsync(string owner, bool useCache = true)
+        {
+            throw new NotImplementedException("Haven't found a way to implement this yet!");
+            yield break;
+        }
+
+        /// <summary>  
+        /// Retrieves metadata for a specific NuGet package version.  
+        /// If no version is provided in the <paramref name="purl"/>, the latest version is used.  
+        /// </summary>  
+        /// <param name="purl">The <see cref="PackageURL"/> representing the package to retrieve metadata for.</param>  
+        /// <param name="includePrerelease">Indicates whether to include pre-release versions when determining the latest version. Defaults to <c>false</c>.</param>  
+        /// <param name="useCache">Indicates whether to use cached data for the request. Defaults to <c>true</c>.</param>  
+        /// <param name="includeRepositoryMetadata">Indicates whether to include repository metadata in the result. Defaults to <c>true</c>.</param>  
+        /// <returns>A <see cref="NuGetPackageVersionMetadata"/> object containing metadata for the specified package version, or <c>null</c> if no metadata is found.</returns>  
+        /// <remarks>  
+        /// This method first determines the latest version of the package if no version is specified in the <paramref name="purl"/>.  
+        /// It then retrieves metadata for the specified or latest version of the package.  
+        /// </remarks>  
+        protected async Task<NuGetPackageVersionMetadata?> GetNuGetPackageVersionMetadata(PackageURL purl, bool includePrerelease = false, bool useCache = true, bool includeRepositoryMetadata = true)
+        {
+            string? latestVersion = await Actions.GetLatestVersionAsync(purl, includePrerelease: includePrerelease, useCache: useCache);
```

Comment:
Nit: Should we avoid making this call if a version has been provided? This result is thrown away if a version is present right?